pages/GridQuestion.js

```diff
@@ -0,0 +1,195 @@
+    await page.getByLabel("Single Select").nth(1).check();
+    await expect(
+      page.getByText(
+        "Choice Display TypeSelectionRadio ButtonsDropdownSliderDefault SliderRYG"
+      )
+    ).toBeVisible();
+    await expect(
+      page.getByRole("button", { name: "Choose Preset" })
+    ).toBeVisible();
+    await page.getByRole("button", { name: "Choose Preset" }).click();
+    await page.getByText("Select", { exact: true }).click();
+    await page.getByText("Good-Poor").click();
+    await page.getByRole("button", { name: "Choose Preset" }).click();
+    await page.getByLabel("Multi Select").nth(2).check();
+    await page
+      .locator("div:nth-child(2) > div > div:nth-child(3) > button")
+      .first()
+      .click();
+    await page
+      .locator(
+        "div:nth-child(5) > div:nth-child(4) > div > div:nth-child(2) > div > div:nth-child(2) > .MuiFormControl-root > .MuiInputBase-root > #text-input-field"
+      )
+      .first()
+      .fill("1");
+    await page
+      .locator(
+        "div:nth-child(5) > div:nth-child(4) > div > div:nth-child(2) > div:nth-child(2) > div:nth-child(2) > .MuiFormControl-root > .MuiInputBase-root > #text-input-field"
+      )
+      .fill("2");
+    await page.getByPlaceholder("Enter option").nth(3).fill("High");
+    await page
+      .locator(
+        "div:nth-child(5) > div:nth-child(4) > div > div:nth-child(2) > div:nth-child(2) > div > .MuiFormControl-root > .MuiInputBase-root"
+      )
+      .first()
+      .click();
+    await page.getByPlaceholder("Enter option").nth(4).fill("Low");
+    await expect(page.getByRole("button", { name: "Save" })).toBeEnabled();
+    await page.getByRole("button", { name: "Save" }).click();
+    await expect(page.locator('.css-uhb5lp')).toHaveScreenshot('Success_dialog_box.png',{maxDiffPixelRatio:0.05});
+    await expect(page.getByRole("heading")).toContainText("Successful");
+    await expect(page.getByLabel("Successful")).toContainText(
+      "The question has been saved successfully."
+    );
+    await expect(page.getByRole("img")).toBeVisible();
+    await expect(page.getByRole("button", { name: "Done" })).toBeVisible();
+    await page.getByRole("button", { name: "Done" }).click();
+    await expect(
+      page.getByRole("button", { name: "Delete" })
+    ).toBeVisible();
+    await expect(
+      page.getByRole("button", { name: "Delete" })
+    ).toBeEnabled();
+    await expect(
+      page.getByRole("button", { name: "Clone" })
+    ).toBeVisible();
+    await expect(
+      page.getByRole("button", { name: "Clone" })
+    ).toBeEnabled();    
+    await expect(page.getByPlaceholder('Enter question', { exact: true })).toHaveValue('Grid Question');
+    await expect(page.getByPlaceholder('Enter row header')).toHaveValue('Row Header');
+    await expect(page.getByPlaceholder('Enter code').first()).toHaveValue('1');
+    await expect(page.getByPlaceholder('Enter code').nth(1)).toHaveValue('2');
+  }
```

Comment:
This function is very big. You can further divide into multiple function, Remember your code should be like a story. People should understand it without too much efforts.